packages/extension-api/src/extension-api.d.ts

```diff
@@ -1339,6 +1346,31 @@ declare module '@podman-desktop/api' {
     readonly location: Uri;
   }
 
+  export type KubernetesGeneratorSelector = KubernetesGeneratorType | ReadonlyArray<KubernetesGeneratorType>;
+
+  export type KubernetesGeneratorType = 'Compose' | 'Pod' | 'Container';
+
+  /**
+   * The result containing a Kubernetes config files
+   */
+  export interface GenerateKubeResult {
+    yaml: string;
+  }
+
+  /**
+   * The KubernetesGeneratorProvider allows an extension to register a custom Kube Generator for a specific
+   * KubernetesGeneratorType.
+   */
+  export interface KubernetesGeneratorProvider {
+    /**
+     * A unique identifier for the provider.
+     */
+    id: string;
```

Comment:
the unique identifier is not provided by the provider (it's generated/handled by Podman desktop)

```suggestion

```